### PR TITLE
add job permission check for data balance

### PR DESCRIPTION
**Pull request by SuperYoko**
## What type of PR is this?
- [x] bug
- [ ] feature
- [ ] enhancement

#### Description:

Users except GUEST can do data balance.

## How do you solve it?

Add data balance permission check for writespace.

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Review by liwenhui-soul — APPROVED** [review]

File: src/graph/service/PermissionCheck.cpp [hunk 1]
```diff
@@ -109,8 +109,19 @@ namespace graph {
...
-    case Sentence::Kind::kDeleteEdges:
+    case Sentence::Kind::kDeleteEdges: {
+      return PermissionManager::canWriteData(session, vctx);
+    }
     case Sentence::Kind::kAdminJob: {
+      auto *adminJobSentence = dynamic_cast<AdminJobSentence *>(sentence);
+      if (adminJobSentence == nullptr) {
+        // should not happend.
+        LOG(WARNING) << "sentence is not AdminJobSentence";
+        return Status::PermissionError("No permission to write space.");
```
Comment (pengweisong) [inline src/graph/service/PermissionCheck.cpp:120]: should the message be sth like "bad format sentence"?

File: src/graph/service/PermissionCheck.cpp [hunk 2]
```diff
@@ -109,8 +109,19 @@ namespace graph {
...
+      }
+      if (adminJobSentence->needWriteSpace()) {
```
Comment (pengweisong) [inline src/graph/service/PermissionCheck.cpp:122]: what does  `needWriteSpace` mean?

Comment (SuperYoko) [reply src/graph/service/PermissionCheck.cpp:122]: > what does `needWriteSpace` mean? Why only balance job need write space? others like compact will not write?

Yes, It's means space data distribuition level permission needed.( And it's the origin permission needed before refactor to jobmanager.

File: src/graph/service/PermissionCheck.cpp (hunk 1 above)
Comment (SuperYoko) [reply src/graph/service/PermissionCheck.cpp:120]: > should the message be sth like "bad format sentence"?

It's a path should never go in, I wan't use check not null first...  Make sence to me.

File: src/graph/service/PermissionCheck.cpp (hunk 2 above)
Comment (pengweisong) [reply src/graph/service/PermissionCheck.cpp:122]: you may need to update the comment before `PermissionCheck::permissionCheck` to explain this

Comment (SuperYoko) [reply src/graph/service/PermissionCheck.cpp:124]: updated.

**Review by critical27 — APPROVED** [review]

**Review by liwenhui-soul — APPROVED** [review]

**Review by pengweisong — APPROVED** [review]

**Pull request merged**